tests/Query/BuilderTest.php

```diff
@@ -456,7 +513,7 @@ public function testException($class, $message, \Closure $build): void
 
         $this->expectException($class);
         $this->expectExceptionMessage($message);
-        $build($builder);
+        $build($builder)->toMQL();
```

Comment:
Why was `toMQL()` required to throw the exception? Is this specific to the regex construction?

Reply:
The exception is thrown when `toMQL` is called, otherwise it's a simple where close.
I could move all the regex transformation logic into the `where` method to throw earlier.

Reply:
I assume it's not needed for the unsupported operator exceptions and only required for the validation in `compileWhereBasic()`. Is that correct?

I'm just curious as to why this is the first time this came up in the PRs reviewed thus far. I would have expected this regex validation to be similar to what we did previously for dates, but perhaps _those_ instances were relying on validation _before_ anything was appended to the `wheres` property.

It's probably a subtle difference from the user's POV (akin to a logic exception when building the query vs. runtime exception at execution time), but I think a logic exception makes more sense. Assuming the change isn't too involved, I expect it would also allow us to reduce the variance in what is appended to the wheres property.

To that end, is it more maintainable to assemble the BSON Regex object up front instead of during `compileWhereBasic()`?

Reply:
In Laravel query builder, there is several ways to append elements to the `$wheres` list. Instead of hacking on each place, it was easier to make the transformation in `compileWhereBasic`.

Reply:
I see the reasoning in not wanting to go down a rabbit hole trying to plug all of the end points in the base Eloquent API.